integrationtests/src/test/java/org/axonframework/integrationtests/testsuite/student/AbstractStudentTestSuite.java

```diff
@@ -199,29 +200,32 @@ protected <T, R> R sendCommand(T payload, Class<R> expectedResultType) {
     }
 
     protected void verifyStudentName(String id, String name) {
-        UnitOfWork uow = new UnitOfWork();
-        uow.executeWithResult(
-                   context -> stateManager.repository(Student.class, String.class)
-                                          .load(id, context)
-                                          .thenAccept(student -> assertEquals(name, student.entity().getName()))
-           )
+        UnitOfWork uow = aUnitOfWork();
+        uow.executeWithResult(context -> stateManager.repository(Student.class, String.class)
+                                                     .load(id, context)
+                                                     .thenAccept(student -> assertEquals(name,
+                                                                                         student.entity().getName())))
            .join();
     }
 
     protected void verifyStudentEnrolledInCourse(String id, String courseId) {
-        UnitOfWork uow = new UnitOfWork();
-        uow.executeWithResult(
-                   context -> stateManager.repository(Student.class, String.class)
-                                          .load(id, context)
-                                          .thenAccept(student -> assertTrue(student.entity()
-                                                                                   .getCoursesEnrolled()
-                                                                                   .contains(courseId)))
-                                          .thenCompose(v -> stateManager.repository(Course.class, String.class)
-                                                                        .load(courseId, context))
-                                          .thenAccept(course -> assertTrue(course.entity()
-                                                                                 .getStudentsEnrolled()
-                                                                                 .contains(id)))
-           )
+        UnitOfWork uow = aUnitOfWork();
+        uow.executeWithResult(context -> stateManager.repository(Student.class, String.class)
```

Comment:
So...we could also grab the `StateManager` from the `context` parameter here, instead of leaving it as a `protected` field of the test suite, right? :-)